File: Problems/Algorithms/121. Best Time to Buy and Sell Stock/BuySellStock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var profit: Int = 0
        var buy: Int = prices[0]
        
        for (price in prices) {
            profit = maxOf(profit, price-buy)
            buy = minOf(buy, price)
        }
        
        return profit
    }
}

File: Problems/Algorithms/121. Best Time to Buy and Sell Stock/buy_sell_stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut buy = prices[0];
        prices.iter().fold(0, |mut acc, x| {
            acc = acc.max(x-&buy);
            buy = buy.min(*x);
            acc
        })
    }
}

